### ISSUE TITLE
Pre-patch 11.0 lua error.

### ISSUE THREAD

**Issue by TreizeEU** [TekNoLogic/tekticles#23]
I'm getting a new lua error since The War Within pre-patch 11.0 has gone live.

[Tecticles lua error.txt](https://github.com/user-attachments/files/16402521/Tecticles.lua.error.txt)

**Comment by TreizeEU** [comment]
I’ve now fixed all known Retail Lua errors in Tekticles locally, including:

- Invalid SetFont calls due to stricter argument validation
- 
- Deprecated InterfaceOptions / SettingsPanel API usage
- 
- ChatFrame font application changes
- 
- Removed PaperDollTitlesPane reference

After these changes, Tekticles loads cleanly on current Retail with no Lua errors.

If useful, I can provide patched files or a diff. No expectation of maintenance — just sharing the fixes.

**Comment by crabby654** [comment]
> I’ve now fixed all known Retail Lua errors in Tekticles locally, including:
> 
>     * Invalid SetFont calls due to stricter argument validation
> 
>     * Deprecated InterfaceOptions / SettingsPanel API usage
> 
>     * ChatFrame font application changes
> 
>     * Removed PaperDollTitlesPane reference
> 
> 
> After these changes, Tekticles loads cleanly on current Retail with no Lua errors.
> 
> If useful, I can provide patched files or a diff. No expectation of maintenance — just sharing the fixes.

I've been using this addon broken for what feels like decades haha, I would love it if you could post your fixes!

**Comment by TreizeEU** [comment]
Sure 🙂

(TL;DR at bottom)

Below are the minimal changes I made to get Tekticles loading cleanly on current Retail with no Lua errors.
These are all fixes for Blizzard API changes / stricter validation; no functional changes beyond that.

sample.lua

Settings panel

Replaced deprecated InterfaceOptions_AddCategory usage with the modern Retail Settings API (Settings.RegisterCanvasLayoutCategory).

Font templates

Guarded font creation so missing templates (e.g. QuestFontHighlight) fall back to a safe default (GameFontNormal).

tekticles.lua

SetFont helper

Modern WoW no longer accepts nil or invalid values for SetFont / SetAlpha.

Updated the helper to:

Never pass nil as the font flags argument

Only call SetAlpha() when the value is between 0 and 1

local function SetFont(obj, font, size, style, r, g, b, sr, sg, sb, sox, soy)
    obj:SetFont(font, size, style or "")

    if sr and sg and sb then
        obj:SetShadowColor(sr, sg, sb)
    end
    if sox and soy then
        obj:SetShadowOffset(sox, soy)
    end
    if r and g and b then
        obj:SetTextColor(r, g, b)
    elseif type(r) == "number" and r >= 0 and r <= 1 then
        obj:SetAlpha(r)
    end
end


ChatFrame fonts

ChatFrame objects now require a valid 3rd argument for SetFont.

f:SetFont(NORMAL, size, "")


Removed obsolete Retail code

Guarded / removed the old PaperDollTitlesPane reference, which no longer exists on Retail.

After these changes, Tekticles loads cleanly on current Retail with no Lua errors (tested with BugGrabber/BugSack).

I’m not maintaining a fork at the moment, but feel free to turn this into a PR or repackage it if useful.



TL;DR – copy/paste fixes (Retail)


sample.lua

1) Settings panel (replace old API call)
Replace the line (at bottom of file) that uses:

InterfaceOptions_AddCategory(frame)


with:

local category = Settings.RegisterCanvasLayoutCategory(frame, "Tekticles")
Settings.RegisterAddOnCategory(category)


2) Font creation (prevent missing font template errors) (line 93)
Replace:

local fs = frame:CreateFontString(nil, "ARTWORK", font)


with:

local fs
if type(_G[font]) == "table" and _G[font].GetFont then
	fs = frame:CreateFontString(nil, "ARTWORK", font)
else
	fs = frame:CreateFontString(nil, "ARTWORK", "GameFontNormal")
end

tekticles.lua

3) Replace the SetFont helper function (top of file)
Replace the entire existing SetFont function with:

local function SetFont(obj, font, size, style, r, g, b, sr, sg, sb, sox, soy)
	obj:SetFont(font, size, style or "")

	if sr and sg and sb then
		obj:SetShadowColor(sr, sg, sb)
	end
	if sox and soy then
		obj:SetShadowOffset(sox, soy)
	end
	if r and g and b then
		obj:SetTextColor(r, g, b)
	elseif type(r) == "number" and r >= 0 and r <= 1 then
		obj:SetAlpha(r)
	end
end


4) ChatFrame font fix (what now should be line 98)
Replace:

f:SetFont(NORMAL, size)


with:

f:SetFont(NORMAL, size, "")


5) Remove or guard obsolete Retail code (now line 103)
Remove (or comment out with -- symbol before the lines) or wrap in an if check the block that references:

PaperDollTitlesPane


(it no longer exists on Retail and causes Lua errors)

After these replacements, Tekticles loads cleanly on current Retail with no Lua errors.

**Comment by crabby654** [comment]
Awesome, thank you! Testing it out ingame and now it seems to work great. The options panel is blank that shows the demo fonts but its not a big deal really. I might mess around with it.
